File: UD-04/P02-Encordados-MongoDB-JPA-Java/src/main/java/es/joseluisgs/p02encordadosmongodbjpa/repositories/RaquetasRepository.java
package es.joseluisgs.p02encordadosmongodbjpa.repositories;

import es.joseluisgs.p02encordadosmongodbjpa.models.Raqueta;
import org.bson.types.ObjectId;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface RaquetasRepository extends MongoRepository<Raqueta, ObjectId> {
    Raqueta findByMarca(String marca);
}

File: UD-04/P03-Encordados-MongoDB-JPA-Kotlin/src/main/kotlin/es/joseluisgs/p02encordadosmongodbjpakotlin/repositories/RaquetasRepository.kt
package es.joseluisgs.p02encordadosmongodbjpakotlin.repositories

import es.joseluisgs.p02encordadosmongodbjpakotlin.models.Raqueta
import org.bson.types.ObjectId
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface RaquetasRepository : MongoRepository<Raqueta, ObjectId> {
    fun findByMarca(marca: String): Raqueta?
}